docs/docs/02-concepts/01-preflight-and-inflight.md

```diff
@@ -334,6 +334,83 @@ Within the first clause of the `lift` block, a list of qualifications on preflig
 Statements within a `lift` block are exempt from the compiler's analyzer that tries to determine preflight object usage automatically.
 If an inflight method is directly or indirectly called within a `lift` block without sufficient resource qualifications, it may result in errors at runtime.
 
+## Building liftable abstractions
+
+Compute environments where inflight code is executed are called **inflight hosts**.
+For example, AWS Lambda, Fly.io machines, and Fargate containers are all examples of compute environments that can run inflight code.
+
+In order to run a piece of inflight code, the inflight host may need to be set up in a certain way.
+For example, to run an inflight function that reads from a storage bucket, the inflight host may require permissions to read from the bucket, or may require certain environment variables to be set, or may require network policies must be configured.
+
+Preflight classes can be used to encapsulate these requirements.
+Any preflight class can implement a method named `onLift` that is called when the class is used in inflight code, where requirements can be added to the inflight host.
```

Comment:
Do we have any more info on onLift where we can link to?

Reply:
I think so -- I just pushed a change so I think we should be generating some API reference docs where we specify the `onLift` method in the SDK. So there's another place where folks can refer to the method signature / find it in the docs.

I should mention there might be an issue if we try to do the same for `onLiftType`, because the method needs to be implemented right now as a static method. But it's not possible to define an interface that enforces a class implements a static method. For now I'm ok looking over this gap 😅